addons/web_editor/static/src/components/html_history_dialog/html_history_dialog.js

```diff
@@ -0,0 +1,124 @@
+/** @odoo-module **/
+
+import { Dialog } from '@web/core/dialog/dialog';
+import { useService } from "@web/core/utils/hooks";
+import {Component, onMounted, useState, markup} from '@odoo/owl';
+import { _t } from 'web.core';
+const { DateTime } = luxon;
+
+class HtmlHistoryDialog extends Component {
+
+    state = useState({
+        latestHistoryDiffs: [],
+        currentHtml: null,
+        currentComparisonHtml: null,
+        currentId: null,
+        buttonDisabled: true,
+    });
+    setup() {
+        this.size = 'xl';
+        this.title = _t("Recent History");
+        this.orm = useService("orm");
+
+        this.comparisonCache = {};
+        this.htmlCache = {};
+
+        onMounted(() => this.init());
+    }
+
+    async init(limit=100) {
+        // limit the number of record and order them by id desc
+        const recIds = this.props.recordsIds.slice(0, limit).reverse();
+
+        // get record data
+        this.state.latestHistoryDiffs = await this.orm.call(
+            'field.html.history.diff',
+            'read',
+            [recIds, ["id", "create_date", "create_uid"]]
+        );
+
+        // get the comparison html for the latest diff
+        this.state.buttonDisabled = false;
+        await this.updateCurrentComparisonHtml(this.state.latestHistoryDiffs[0].id);
+    }
+
+    async updateCurrentComparisonHtml(historyDiffId) {
+        if (this.state.currentId === historyDiffId || this.state.buttonDisabled) {
+            return;
+        }
+        this.state.buttonDisabled = true;
+        this.state.currentId = historyDiffId;
+        this.state.currentComparisonHtml = await this.getComparisonAtDiffId(historyDiffId);
+        this.state.currentHtml = await this.getHtmlAtDiffId(historyDiffId);
+        this.state.buttonDisabled = false;
+    }
+
+    async getComparisonAtDiffId(historyDiffId) {
+        if (!this.comparisonCache[historyDiffId]) {
+            const comparison = await this.orm.call(
+                "field.html.history.diff",
+                "get_comparison",
+                [historyDiffId]
+            );
+            this.comparisonCache[historyDiffId] = markup(comparison);
+        }
+        return this.comparisonCache[historyDiffId];
+    }
+    async getHtmlAtDiffId(historyDiffId) {
+        if (!this.htmlCache[historyDiffId]) {
```

Comment:
I'm not sure I understand the use of the cache... ah, it's to not recall the server a second time for the same result, right ? I think there is a `memoize` util in web for that so you don't have to handle the caching manually (since there doesn't seem to be any cache invalidation mechanism involved.)

Reply:
I'll look into that